File: archives/C#/0303. Range Sum Query - Immutable.cs
public class NumArray {
    
    IList<int> rep=new List<int>();
    public NumArray(int[] nums) {
        int sumNums=0;
        for(int i=0;i<nums.Length;i++){
            rep.Add(sumNums);
            sumNums+=nums[i];
        }
        rep.Add(sumNums);
    }
    
    public int SumRange(int i, int j) {
        return rep[j+1]-rep[i];
    }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.SumRange(i,j);
 */

File: archives/20190519python/0303. Range Sum Query - Immutable.py
class NumArray(object):

    def __init__(self, nums):
        """
        :type nums: List[int]
        """
        self.sum_nums=[]
        rep=0
        for num in nums:
            rep+=num
            self.sum_nums.append(rep)

    def sumRange(self, i, j):
        """
        :type i: int
        :type j: int
        :rtype: int
        """
        if i==0:
            return self.sum_nums[j]
        return self.sum_nums[j]-self.sum_nums[i-1]
        


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(i,j)
